fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractDimensionRowProvider.java

```diff
@@ -0,0 +1,195 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.application;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary;
+import com.yahoo.bard.webservice.data.dimension.DimensionRow;
+import com.yahoo.bard.webservice.druid.client.FailureCallback;
+import com.yahoo.bard.webservice.druid.client.HttpErrorCallback;
+import com.yahoo.bard.webservice.druid.model.datasource.DataSource;
+import com.yahoo.bard.webservice.druid.model.datasource.TableDataSource;
+import com.yahoo.bard.webservice.table.PhysicalTableDictionary;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * The AbstractDimensionRowProvider provides a way to load dimension rows onto dimensions.
+ */
+public abstract class AbstractDimensionRowProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractDimensionRowProvider.class);
+
+    private final List<Dimension> dimensions;
+    private final List<DataSource> dataSources;
+
+    private HttpErrorCallback errorCallback;
+    private FailureCallback failureCallback;
+    private BiConsumer<Dimension, DimensionRow> dimensionRowConsumer;
+    private Consumer<Dimension> dimensionLoadedConsumer;
```

Comment:
Alternatively I could just pull the two methods that are actually used from `DimensionLoader` into `AbstractDimensionRowProvider` since they're unlikely to change

```java
@Override
public void accept(Dimension dimension, DimensionRow dimensionRow) {
    dimension.addDimensionRow(dimensionRow);
}

@Override
public void accept(Dimension dimension) {
    // Tell the dimension it's been updated
    dimension.setLastUpdated(DateTime.now());
}
```